src/argocd/ArgoCDServer.ts

```diff
@@ -109,4 +109,32 @@ export class ArgoCDServer {
     // Return the release tag without the build metadata (e.g., v2.4.0+91aefab -> v2.4.0).
     return responseJson.Version.split('+')[0];
   }
+
+  async getAppCollectionDiffs(appCollection: AppCollection): Promise<Diff[]> {
+    const diffs: Diff[] = [];
+
+    // Settle all promises in parallel.
+    let appCollectionDiffPromises: Promise<Diff>[] = [];
+    appCollection.apps.forEach(app => {
+      appCollectionDiffPromises.push(this.getAppLocalDiff(app));
+    });
+    await Promise.allSettled(appCollectionDiffPromises).then(results => {
+      (
+        results.filter(result => result.status === 'fulfilled') as PromiseFulfilledResult<Diff>[]
+      ).forEach(result => {
+        let appDiff = result.value;
+
+        if (appDiff.error) {
+          core.setFailed(`ArgoCD diff failed for Application '${appDiff.app.metadata.name}'`);
+        } else if (appDiff.diff != '') {
+          core.info(`Found diff for Application '${appDiff.app.metadata.name}'.`);
+          diffs.push(appDiff);
+        } else {
+          core.debug(`No diff found for Application '${appDiff.app.metadata.name}'.`);
+        }
+      });
+    });
```

Comment:
I'm finding this block of code is really hard to read with the formatting & nesting.

I'm not positive this suggestion actually works, but in theory it can be made more readable by doing something like this:

```suggestion
    const results = await Promise.allSettled(appCollectionDiffPromises);
    const filteredResults = results.filter(result => result.status === 'fulfilled') as PromiseFulfilledResult<Diff>[];
    
    filteredResults.forEach(result => {
        let appDiff = result.value;

        if (appDiff.error) {
          core.setFailed(`ArgoCD diff failed for Application '${appDiff.app.metadata.name}'`);
        } else if (appDiff.diff != '') {
          core.info(`Found diff for Application '${appDiff.app.metadata.name}'.`);
          diffs.push(appDiff);
        } else {
          core.debug(`No diff found for Application '${appDiff.app.metadata.name}'.`);
        }
    });
```

In fact I'm almost positive the `const results = await Promise.allSettled(appCollectionDiffPromises);` won't work

Reply:
Thoughts on having one step be getting the results (including filtering) and the other to act on the results. This seems easy enough to read.
```suggestion
    let results = (await Promise.allSettled(appCollectionDiffPromises)).filter(
      result => result.status === 'fulfilled'
    ) as PromiseFulfilledResult<Diff>[];

    results.forEach(result => {
      let appDiff = result.value;
      if (appDiff.error) {
        core.setFailed(`ArgoCD diff failed for Application '${appDiff.app.metadata.name}'`);
      } else if (appDiff.diff != '') {
        core.info(`Found diff for Application '${appDiff.app.metadata.name}'.`);
        diffs.push(appDiff);
      } else {
        core.debug(`No diff found for Application '${appDiff.app.metadata.name}'.`);
      }
    });
```

Reply:
LGTM, looks much better